### DISCUSSION TITLE
Unable to execute weed filer.copy in a nodejs script?

### DISCUSSION THREAD

**Discussion by hunglsxx** [seaweedfs/seaweedfs#3232] [Q&A]
Hi guys,
When I execute this command `/path/to/weed filer.copy /path/to/file.mp4 http://localhost:8888/path/to/file/` everything works great.
<img width="589" alt="image" src="https://user-images.githubusercontent.com/50700597/175767040-f1fc404d-09de-4ebe-8c30-f166eb3e7feb.png">
<img width="605" alt="image" src="https://user-images.githubusercontent.com/50700597/175767062-18aa044a-f9a2-4578-9fa9-3d146a2543f5.png">

But when I include it in my script like this
<img width="331" alt="image" src="https://user-images.githubusercontent.com/50700597/175766914-07b0517e-f2ea-4b84-aa6b-cd0c7e1f1034.png">
It doesn't seem to work
With the result returned
<img width="331" alt="image" src="https://user-images.githubusercontent.com/50700597/175766939-b783ef33-c67d-483c-bba9-b7df050fe924.png">
I don't understand why weed filter.copy doesn't work this way, please help

**Comment by chrislusf** [comment]
maybe javascript has some limitation on multi-threading? Not sure. You can try to find the `weed filer.copy` process and run `kill -3 <pid>` to get a thread dump.

**Reply by hunglsxx** [reply]
Yes, i'm thinking about that, i will try your recommend. Thank you so much!

**Comment by hunglsxx** [comment]
Thank @chrislusf 
I used `spawn` or `execFile` method and everything worked.
I think this happens because the exec method in nodejs creates a shell to execute the command, and `spawn` and `execFile` are created directly under a new process.
